### ISSUE TITLE
vector2 changes to float

### ISSUE THREAD

**Issue by duane-r** [godotengine/godot#41725]
**Godot version:**
v4.0.dev.custom_build.4dd915028

**OS/device including version:**
Archlinux with openbox window manager

**Issue description:**
When a vector2 is copied from an array and back, it can change to a float, but only under weirdly specific circumstances. This is the relevant code (a small part of a bunnymark):

```gdscript
	# If you use a variable index instead of a for loop -- it works.
	for i in a.size():

		var pos: Vector2 = Vector2.ZERO
		var speed: Vector2 = a[i]
		pos.y += 10 * delta

		# If you comment these lines out -- it works.
		if pos.y > 990:
			pos.y = 990
		if pos.y < 0:
			pos.y = 0

		a[i] = speed
		if not a[i] is Vector2:
			print(to_json(a[i]))
		assert(a[i] is Vector2)
```

Note that speed is not being changed anywhere. This error is so odd that I'm almost convinced that I'm doing something wrong. If you replace the array with a dictionary, the problem goes away.

**Steps to reproduce:**
Run the attached sample project. It requires very specific conditions. The variable "speed" should always be a vector2, but it changes into a float and the assertion fails. This happens reliably, despite the strange conditions needed to trigger it.

**Minimal reproduction project:**

[variable-bug.zip](https://github.com/godotengine/godot/files/5164799/variable-bug.zip)

Edit: correct bad hash

**Comment by groud** [comment]
What is 38c2524cd? Have you tried with latest master or a stable version?

**Comment by ThakeeNathees** [comment]
same issue as #42050 and fixed here: #42067. see: https://github.com/godotengine/godot/issues/42050#issuecomment-692046984
